test/http/basic_parser_v1.cpp

```diff
@@ -481,6 +529,87 @@ class basic_parser_v1_test : public beast::detail::unit_test::suite
         parse_ev<true>("GET / HTTP/1.1\r\nf :", parse_error::bad_field);
     }
 
+    void testInvalidMatrix()
+    {
+        using boost::asio::buffer;
+        std::string s;
+        std::size_t n;
+
+        for(n = 0;; ++n)
```

Comment:
Nit: Move the decl of `n` into the loop and remove the one at the outer scope: `for(size_t n=0;....)`.